code/game/machinery/atmos_exporter.dm

```diff
@@ -0,0 +1,278 @@
+/obj/machinery/atmos_points
+	name = "atmospheric export computer"
+	desc = "used to monitor the galactic atmos markets"
+	icon = 'icons/obj/machines/engi_points.dmi'
+	icon_state = "atmos"
+	verb_say = "states"
+	density = 1
+	anchored = 1
+	var/market_cooldown = 0
+	var/list/quality = list("Economy","Standard","Premium")
+	var/export_target = ""
+	var/export_name = "------"
+
+	var/list/gas = list()
+	var/upressure = 0
+	var/lpressure = 0
+	var/utemp = 0
+	var/ltemp = 0
+	var/price = 200
+	var/list/concentration = list()
+	var/can_turf
+
+	var/economy_name = ""
+	var/list/e_gas = list() // budget stats
+	var/e_pressure = 0
+	var/e_temp = 0
+	var/e_price = 200
+	var/e_concentration1 = 0
+
+	var/standard_name = ""
+	var/list/s_gas = list() // standard stats
+	var/s_pressure = 0
+	var/s_temp = 0
+	var/s_price = 600
+	var/s_concentration1 = 0
+	var/s_concentration2 = 0
+
+	var/premium_name = ""
+	var/list/p_gas = list() // premium stats
+	var/p_pressure = 0
+	var/p_temp = 0
+	var/p_price = 1000
+	var/p_concentration1 = 0
+	var/p_concentration2 = 0
+	var/p_concentration3 = 0
+
+/obj/machinery/atmos_points_exporter
+	name = "atmospheric canister exporter"
+	desc = "used to export canisters for engineering points"
+	icon = 'icons/obj/machines/teleporter.dmi'
+	icon_state = "floorflush_c"
+	verb_say = "states"
+	density = 0
+	anchored = 1
+
+/obj/machinery/atmos_points_exporter/proc/ship()
+	for(var/obj/machinery/portable_atmospherics/canister/C in get_turf(src))
+		icon_state = "floorflush_a"
+		playsound(src, 'sound/machines/Ding.ogg', 100, 1)
+		spawn(20)
+			animate(C, transform = matrix() - matrix(), alpha = 0, color = rgb(0, 0, 0), time = 10)
+			for(var/t in 1 to 5)
+				if(!C || QDELETED(C))

```

Comment:
after any spawn and sleep one must check if you are qdeleted or any of the things you're operating on are qdeleted, i.e check src as well